packages/block-editor/src/components/iframe/use-scale-canvas.js

```diff
@@ -319,12 +319,11 @@ export function useScaleCanvas( {
 		const trigger =
 			iframeDocument && previousIsZoomedOut.current !== isZoomedOut;
 
-		previousIsZoomedOut.current = isZoomedOut;
-
 		if ( ! trigger ) {
 			return;
 		}
 
+		previousIsZoomedOut.current = isZoomedOut;
```

Comment:
This is testing well for me. I think I'd prefer to keep the previous `previousIsZoomedOut.current !== isZoomedOut` alongside the setting of the ref though. I think it will be easier to understand why we don't want the animation to run when the iframeDocument is false. What do you think of this instead?

```
// If we don't have an iframe document, we don't run or track the animation.
if ( ! iframeDocument ) {
	return;
}

const triggerAnimation = isZoomedOut !== previousIsZoomedOut.current;
previousIsZoomedOut.current = isZoomedOut;

// Zoom state has not changed, so we don't need to run the animation.
if ( ! triggerAnimation ) {
	return;
}
```

Reply:
Updated in 0dbaf3821b5ae42afca9d0c2661d4225d235e7f2

The suggested logic seems to work correctly!